python-threatexchange/threatexchange/cli/config_cmd.py

```diff
@@ -527,15 +531,88 @@ def get_name(cls) -> str:
 
     @classmethod
     def init_argparse(cls, settings: CLISettings, ap: argparse.ArgumentParser) -> None:
-        ap.add_argument(
+        import_cmds = ap.add_mutually_exclusive_group()
+        import_cmds.add_argument(
+            "--list-available-collabs",
+            "-L",
+            action="store_true",
+            help="query the API to list available collabs",
+        )
+        import_cmds.add_argument(
+            "--import-collab",
+            "-I",
+            type=int,
+            help="import a collaboration by privacy group ID",
+        )
+
+        # Not actually a type of import cmd, but to add exclusivity logic
+        config_cmds = import_cmds.add_argument_group()
+        config_cmds.add_argument(
             "--api-token",
-            help="set the default api token",
+            help="set the default api token (https://developers.facebook.com/tools/accesstoken/)",
         )
 
-    def __init__(self, api_token: t.Optional[str]) -> None:
+    def __init__(
+        self,
+        api_token: t.Optional[str],
+        list_available_collabs: bool,
+        import_collab: t.Optional[int],
+    ) -> None:
         self.api_token = api_token
+        self.action = self.execute_config
+        if list_available_collabs:
+            self.action = self.execute_list_collabs
+        elif import_collab is not None:
+            tmp_for_typing = import_collab
+            self.action = lambda s: self.execute_import(s, tmp_for_typing)
```

Comment:
It actually cannot! mypy claims that it's the wrong type (it didn't seem to properly inherit the `not None`-ness of the variable unless I re-assigned it. No earthly idea why, since I've seen similar blocks pass in other cases.